torch/csrc/jit/script/init.cpp

```diff
@@ -309,23 +309,20 @@ void initJitScriptBindings(PyObject* module) {
       .def("_get_modules", [](Module& self) -> py::tuple {
         auto & modules = self.get_modules();
         py::tuple result(modules.size());
-        for(size_t i = 0; i < modules.size(); ++i) {
-          auto & nm = modules[i];
-          result[i] = std::make_pair(nm.name, nm.module);
+        size_t index = 0;
+        for (auto& item : self.get_modules()) {
+          result[index++] = std::make_pair(item.key, item->module);
         }
         return result;
       })
       .def("_get_parameters", [](Module& self) -> py::tuple {
-        auto & parameters = self.get_parameters();
-        py::tuple result(parameters.size());
-        for(size_t i = 0; i < parameters.size(); ++i) {
-          auto & p = parameters[i];
-          py::tuple r(3);
-          result[i] = std::make_tuple(
-            p.name,
-            static_cast<const autograd::Variable&>(*p.slot()),
-            p.is_buffer);
-
+        py::tuple result(self.get_parameters().size());
+        size_t index = 0;
+        for (auto& parameter : self.get_parameters()) {
```

Comment:
Can we please use regular `for` loops when we iterate over multiple things? I find the mutable `index` pattern really annoying, as it only works if you iterate over 2 things. I'd stick to a rule of 1 item = range-based, 2+ items = old-style for loop.

Reply:
ok sounds good